src/main/java/com/jcabi/github/mock/MkHook.java

```diff
@@ -104,6 +105,37 @@ public int number() {
     }
     @Override
     public JsonObject json() throws IOException {
-        throw new UnsupportedOperationException("#json()");
+        final XML hook = this.storage.xml().nodes(
+            String.format(
+                "//repo[@coords='%s']/hooks/hook[id='%s']",
+                this.coords, this.num
+            )
+        ).get(0);
+        final JsonArrayBuilder events = Json.createArrayBuilder();
+        for (final XML event : hook.nodes("events/child::*")) {
+            // @checkstyle MultipleStringLiteralsCheck (1 line)
+            events.add(event.xpath("name()").get(0));
+        }
+        final JsonObjectBuilder configs = Json.createObjectBuilder();
+        for (final XML config : hook.nodes("configs/child::*")) {
+            configs.add(
+                // @checkstyle MultipleStringLiteralsCheck (1 line)
+                config.xpath("name()").get(0), config.xpath("text()").get(0)
+            );
+        }
+        return Json.createObjectBuilder()
+            .add("id", this.number())
+            .add("url", hook.xpath("url/text()").get(0))
+            .add("test_url", hook.xpath("test_url/text()").get(0))
+            .add("ping_url", hook.xpath("ping_url/text()").get(0))
+            .add("name", hook.xpath("name/text()").get(0))
+            .add("events", events)
+            .add(
+                "active",
+                Boolean.parseBoolean(hook.xpath("active/text()").get(0))
+            ).add("config", configs)
+            .add("updated_at", hook.xpath("updated_at/text()").get(0))
+            .add("created_at", hook.xpath("created_at/text()").get(0))
+            .build();
```

Comment:
@llorllale I don't like the fact that we cherrypick attributes from the XML and create a JsonObject only with them. I think this method should have one line: 

```java
return new JsonNode(xml).json(
    this.storage.xml().nodes(
        String.format(
            "//repo[@coords='%s']/hooks/hook[id='%s']",
            this.coords, this.num
        )
    ).get(0)
);
```

So, simply parse the XML content into JSON and let ``MkHooks.create(...)`` worry about creating the Hook with the right attributes/contents.

Could you do this?

Reply:
@amihaiemil I was not aware of/didn't remember `JsonNode`. I agree with you, and also think it would clean up the tests *a lot*.